spec/spec_helper.rb

```diff
@@ -14,6 +14,10 @@
 require 'vcr'
 require 'cgi'
 
+# Fail tests that try to include stuff in `main`
+require_relative 'support/test_contamination'
+Spec::Support::TestContamination.setup(self)
```

Comment:
@chrisarcand Not sure if there is a case where we'd want to call this with anything else.  However, we need to pass `self` in because it's an instance of some anonymous class, not `Object, Kernel, BasicObject, TOPLEVEL_BINDING or main`.